src/main/java/org/cbioportal/service/impl/ClinicalEventServiceImpl.java

```diff
@@ -108,4 +122,101 @@ public List<ClinicalEventTypeCount> getClinicalEventTypeCounts(List<String> stud
             .map(e -> new ClinicalEventTypeCount(e.getKey(),e.getValue()))
             .collect(Collectors.toList());
     }
+
+    @Override
+    public List<ClinicalData> getSurvivalData(List<String> studyIds,
```

Comment:
This needs to be broken down into smaller functions.

Reply:
updated